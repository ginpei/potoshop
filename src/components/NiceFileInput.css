.NiceFileInput {
}
.NiceFileInput-input {
  display: none;
}
