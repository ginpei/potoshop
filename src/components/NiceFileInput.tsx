/**
 * @example
 * <NiceFileInput>Nice!</NiceFileInput>
 * @example
 * <NiceFileInput icon="thumbs-up">Nice!</NiceFileInput>
 * @example
 * <NiceFileInput icon="fa-thumbs-up">Nice!</NiceFileInput>
 * @example
 * <NiceFileInput onClick={onClick}>Nice!</NiceFileInput>
 */

import * as React from 'react';
import './NiceFileInput.css';

interface INiceFileInputProps {
  accept?: string;
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
}
interface INiceFileInputState {
}

class NiceFileInput extends React.Component<INiceFileInputProps, INiceFileInputState> {
  constructor (props: INiceFileInputProps) {
    super(props);
    this.state = {
    };
    this.onChange = this.onChange.bind(this);
  }

  public render () {
    return (
      <label className="NiceFileInput niceButtonBase">
        <i className="fa fa-cloud-upload niceButtonBase-leftIcon" aria-hidden="true"/>
        Select file
        <input className="NiceFileInput-input" type="file"
          accept={this.props.accept}
          onChange={this.onChange}
          />
      </label>
    );
  }

  protected onChange (event: React.ChangeEvent<HTMLInputElement>) {
    this.props.onChange!(event);
  }
}

export default NiceFileInput;
