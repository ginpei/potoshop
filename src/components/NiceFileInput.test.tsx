import * as enzyme from 'enzyme';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';
import NiceFileInput from './NiceFileInput';

// possible to move to test.js?
enzyme.configure({ adapter: new Adapter() });

describe('<NiceFileInput/>', () => {
  it('renders', () => {
    const wrapper = shallow<NiceFileInput>(
      <NiceFileInput/>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('applies accept attr', () => {
    const wrapper = shallow<NiceFileInput>(
      <NiceFileInput
        accept="image/*"
        />,
    );
    expect(wrapper.find('input').prop('accept')).toBe('image/*');
  });

  it('invokes onChange callback', () => {
    const onChange = jest.fn();
    const wrapper = shallow<NiceFileInput>(
      <NiceFileInput
        onChange={onChange}
        />,
    );
    wrapper.find('input').simulate('change');
    expect(onChange).toBeCalled();
  });
});
