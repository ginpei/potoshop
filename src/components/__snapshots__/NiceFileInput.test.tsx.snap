// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NiceFileInput/> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NiceFileInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <i
          aria-hidden="true"
          className="fa fa-cloud-upload niceButtonBase-leftIcon"
        />,
        "Select file",
        <input
          className="NiceFileInput-input"
          onChange={[Function]}
          type="file"
        />,
      ],
      "className": "NiceFileInput niceButtonBase",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "className": "fa fa-cloud-upload niceButtonBase-leftIcon",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      "Select file",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "accept": undefined,
          "className": "NiceFileInput-input",
          "onChange": [Function],
          "type": "file",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <i
            aria-hidden="true"
            className="fa fa-cloud-upload niceButtonBase-leftIcon"
          />,
          "Select file",
          <input
            className="NiceFileInput-input"
            onChange={[Function]}
            type="file"
          />,
        ],
        "className": "NiceFileInput niceButtonBase",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-hidden": "true",
            "className": "fa fa-cloud-upload niceButtonBase-leftIcon",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "Select file",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "accept": undefined,
            "className": "NiceFileInput-input",
            "onChange": [Function],
            "type": "file",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
